insights/src/main/kotlin/com/algolia/instantsearch/insights/webservice/Extensions.kt

```diff
@@ -0,0 +1,47 @@
+package com.algolia.instantsearch.insights.webservice
+
+import com.algolia.instantsearch.insights.InsightsLogger
+import com.algolia.instantsearch.insights.database.Database
+import com.algolia.instantsearch.insights.event.Event
+import com.algolia.instantsearch.insights.event.EventResponse
+
+
+internal fun List<EventResponse>.filterEventsNetworkFailure(): List<EventResponse> {
+    return this.filter { it.code == -1 }
+}
+
+internal fun Int.isValidHttpCode() = this == 200 || this == 201
```

Comment:
Is there a reason why not `inlining` these functions? :thinking:

Reply:
<img width="1117" alt="screen shot 2018-06-28 at 10 36 45" src="https://user-images.githubusercontent.com/3338489/42023040-32bdfb4a-7abf-11e8-8c87-e24c40b60c27.png">

What we can do though, is this:

```kotlin
internal inline val Int.isValidHttpCode get() = this == 200 || this == 201
```

What do you think ?

Reply:
I think I like best the idea of having a single worker. They are costly in resources, so it would really makes sense to have a single worker handling multiple indices, which would then mitigate the need of being able to delete a worker in the first place.

Reply:
On inlining: I would default to inlining helper functions to avoid method fixed costs (dex size, stack jump), but maybe my understanding of inlining is limited. What's the benefit of not inlining by default? Do you feel such method call costs are better than bytecode duplication costs?